docs/source/package_definition.rst

```diff
@@ -113,6 +113,12 @@ implicit :attr:`this` object:
    Do not reference other early bound or late bound attributes in
    your early bound function. An error will be raised if you do.
 
+.. warning::
+   Certain package attributes cannot be accessed inside of an early bound function such as :attr:`this.root`.
+
+   Additionally, :ref:`build environment variables <build-environment-variables>` cannot be accessed inside
```

Comment:
I feel like we should clarify or add details to the first sentence in the early section instead of adding a warning here.

```
Early binding functions use the `@early` decorator. They are evaluated at *build time*, hence the
'early' in 'early binding'. Any package attribute can be implemented as an early binding function.
```
This could be expanded a little bit to clarify what "at build time" means. "build time" here is very subtle and means that it's evaluated outside a resolve. In other words, they are evaluated before a resolve happens. Semantics here are very important.